### ISSUE TITLE
Supports unified management of heterogeneous clusters

### ISSUE THREAD

**Issue by wxiaoke** [pingcap/tidb-operator#4774]
## Feature Request

**Is your feature request related to a problem? Please describe:**

when deploying a heterogeneous cluster, need to upgrade the whole cluster. 
The rolling update order of components must be manually controlled, like: pd --> TiFlash--> TiKV -->TiDB

https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-heterogeneous-tidb-cluster#deploy-a-heterogeneous-cluster

**Describe the feature you'd like:**

If we have a CRD like TiDBCluster Group , to manger the heterogeneous clusters will be fine.

When need to upgrade, just edit the group and the program will automatically control the order.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by mikechengwei** [comment]
Why does deploying a heterogeneous cluster trigger an origin cluster update?

**Comment by wxiaoke** [comment]
For example: If I have 3 clusters :cluster1(3pd+6TiKV+6TiFlash+2TiDB),cluster2(6TiKV+2TiFlash+2TiDB), cluster3(3TiDB)
cluster1 is the origin cluster , cluster2 and cluster3 are heterogeneous clusters based on cluster1.

when I need to upgrade the whole clusters, need to manually controll the order of components.

**Comment by mikechengwei** [comment]
Good idea.We can create a crd to orchestrate origin clusters and heterogeneous clusters, which can manage existing clusters.

**Comment by wxiaoke** [comment]
Yes

**Comment by mikechengwei** [comment]
However, previously heterogeneous clusters were used to deploy tidb across multiple k8s clusters to form a cluster. Now this design may be limited to a k8s cluster.
